vcs/gitcmd/repo.go

```diff
@@ -221,6 +235,30 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 	return branches, nil
 }
 
+func (r *Repository) branchesWithCommit(commitID string) (map[string]bool, error) {
+	cmd := exec.Command("git", "branch", "--contains", commitID)
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(out), "\n")
+	branches := make([]string, 0)
+	for _, line := range lines {
+		if len(line) >= 2 {
+			branches = append(branches, line[2:])
+		}
+	}
+
+	branches_ := make(map[string]bool)
```

Comment:
As suggested above, make this a set (value type `struct{}` not `bool`).

Also, `map[string]bool{}` (or the suggested `map[string]struct{}{}`) is shorter than `make`ing it. But here you know the len, so you should call `make(map[string]struct{}, len(branches))`.

Actually - can you omit the construction of the intermediate slice and just create the map directly in the loop above?